libvirt.go

```diff
@@ -113,6 +113,30 @@ func (l *Libvirt) Capabilities() ([]byte, error) {
 	return []byte(caps), err
 }
 
+// called at connection time
+// it calls auth-list and authenticates if such auth type is supported
+func (l *Libvirt) authenticate() error {
+	resp, err := l.AuthList()
+	if err != nil {
+		return err
+	}
+
+loop:
+	for _, auth := range resp {
+		switch auth {
+		case constants.AuthNone:
+			break loop
+		case constants.AuthPolkit:
+			_, err := l.AuthPolkit()
+			if err != nil {
+				return err
+			}
+			break loop
+		}
```

Comment:
nit: this might read slightly better to just add a default case that does a "continue", and then outside the switch just do a break.  Then you don't need multiple break statements or a label.  Just preference though, not something I feel particularly strongly about.
example:  https://play.golang.org/p/GzPizpqHuiV

Reply:
Agreed it is better to read.